src/main/java/com/sforce/ws/transport/JdkHttpTransport.java

```diff
@@ -209,7 +211,13 @@ public InputStream getContent() throws IOException {
             }
         }
 
-        successful = connection.getResponseCode() < 400;
+        statusCode = connection.getResponseCode();
+        successful = statusCode < 400;
+        
+        headers = new HashMap<String, Collection<String>>();
```

Comment:
Why can't you just use the new `HashMap<String, ? extends Collection<String>>(connection.getHeaderFields())`?